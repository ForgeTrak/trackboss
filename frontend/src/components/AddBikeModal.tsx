import React, { useState } from 'react';
import {
    Modal,
    ModalOverlay,
    ModalContent,
    ModalCloseButton,
    Button,
    Divider,
    Heading,
    ModalFooter,
    VStack,
    Input,
    Select,
    NumberInput,
    NumberInputField,
    NumberInputStepper,
    NumberIncrementStepper,
    NumberDecrementStepper,
} from '@chakra-ui/react';

interface modalProps {
  isOpen: boolean,
  onClose: () => void,
  // eslint-disable-next-line no-unused-vars
  addBike: (year: string, make: string, model: string) => void,
}

export default function EditBikesModal(props: modalProps) {
    const [bikeYear, setBikeYear] = useState<string>('');
    const [bikeMake, setBikeMake] = useState<string>('');
    const [bikeModel, setBikeModel] = useState<string>('');

    const handleEditedBikeYear = (event: { target: { value: any; }; }) => setBikeYear(event.target.value);
    const handleEditedBikeMake = (event: { target: { value: any; }; }) => setBikeMake(event.target.value);
    const handleEditedBikeModel = (event: { target: { value: any; }; }) => setBikeModel(event.target.value);
    const thisYear = (new Date()).getFullYear();
    const minModelYear = thisYear - 55;
    const maxModelYear = thisYear + 1;

    return (
        <Modal isCentered size="xl" isOpen={props.isOpen} onClose={props.onClose}>
            <ModalOverlay />
            <ModalContent>
                <Heading pl={2} pr={2} textAlign="center">
                    Add New Bike
                </Heading>
                <Divider mb={5} />
                <ModalCloseButton />
                <VStack
                    align="left"
                    mr={5}
                    ml={5}
                >
                    <NumberInput
                        variant="outline"
                        defaultValue={maxModelYear}
                        min={minModelYear}
                        max={maxModelYear}
                    >
                        <NumberInputField onChange={handleEditedBikeYear} />
                        <NumberInputStepper>
                            <NumberIncrementStepper />
                            <NumberDecrementStepper />
                        </NumberInputStepper>
                    </NumberInput>
                    <Select
                        variant="outline"
                        placeholder="Bike Make"
                        onChange={handleEditedBikeMake}
                        size="md"
                    >
                        <option value="Beta">Beta</option>
                        <option value="Cobra">Cobra</option>
                        <option value="Gas Gas">Gas Gas</option>
                        <option value="GPX">GPX</option>
                        <option value="Honda">Honda</option>
                        <option value="Husqvarna">Husqvarna</option>
                        <option value="Kawasaki">Kawasaki</option>
                        <option value="KTM">KTM</option>
                        <option value="Other">Other</option>
                        <option value="Pitster Pro">Pitster Pro</option>
                        <option value="Suzuki">Suzuki</option>
                        <option value="Yamaha">Yamaha</option>
                    </Select>
                    <Input
                        variant="outline"
                        placeholder="Bike Model"
                        value={bikeModel}
                        onChange={handleEditedBikeModel}
                        size="md"
                    />
                </VStack>
                <ModalFooter>
                    <Button
                        variant="ghost"
                        mr={3}
                        size="lg"
                        onClick={
                            () => {
                                props.onClose();
                            }
                        }
                    >
                        Close
                    </Button>
                    <Button
                        color="green"
                        variant="ghost"
                        mr={3}
                        size="lg"
                        onClick={
                            () => {
                                props.addBike(bikeYear, bikeMake, bikeModel);
                                props.onClose();
                            }
                        }
                    >
                        Save
                    </Button>
                </ModalFooter>
            </ModalContent>
        </Modal>
    );
}
