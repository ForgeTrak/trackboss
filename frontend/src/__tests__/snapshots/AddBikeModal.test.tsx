import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import AddBikeModal from '../../components/AddBikeModal';

function add() {
    // void function
}

function onClose() {
    // void function
}

// Necessary to mock the chakra-ui modal portal, so that the renderer can target the modal instead of the page
const divWithChildrenMock = (children: any, identifier: any) => <div data-testId={identifier}>{children}</div>;
jest.mock('@chakra-ui/react', () => (
    {
        ...jest.requireActual('@chakra-ui/react'),
        PortalManager: jest.fn(({ children }) => divWithChildrenMock(children, 'portal')),
    }
));

describe('add bike modal', () => {
    it('renders correctly', () => {
        const header = renderer.create(
            <BrowserRouter>
                <AddBikeModal isOpen onClose={onClose} addBike={add} />
            </BrowserRouter>,
        ).toJSON();
        expect(header).toMatchSnapshot();
    });
});
