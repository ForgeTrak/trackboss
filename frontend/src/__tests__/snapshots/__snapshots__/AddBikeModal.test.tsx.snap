// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add bike modal renders correctly 1`] = `<span />`;
